more info after the What's new Sessions' questions (#10803)
openvinotoolkit/openvino: docs/OV_Runtime_UG/automatic_batching.md

@endsphinxdirective

> For the *explicit* usage, you can limit the batch size using "BATCH:GPU(4)",  where 4 is the number of requests running in parallel. Notice that since the CPU doesn't support the `ov::optimal_batch_size` yet, it works with the auto-batching only when explicit batch size is specified, for example "BATCH:CPU(16)". 

Reviewer: do we need to write specifically about CPU? Guide must be generic.

`Notice if device doesn't support the `ov::optimal_batch_size` yet, it works with the auto-batching only when explicit batch size is specified, for example "BATCH:<device>(16)"`

 - Use the requests, grouped by the batch size, together. For example, the first 4 requests are inferred, while the second group of the requests is being populated. Essentially, the Automatic Batching shifts the asynchronousity from the individual requests to the groups of requests that constitute the batches.
  - Balance the 'timeout' value vs the batch size. For example in many cases having smaller timeout value/batch size may yield better performance than large batch size, but with the timeout value that is not large enough to accommodate the full number of the required requests. 

The following are limitations of the current implementations:
 - Although less critical for the throughput-oriented scenarios, the load-time with auto-batching increases by almost 2x.
 - Certain networks are not safely reshape-able by the "batching" dimension (specified as 'N' in the layouts terms). Also, if the batching dimension is not zero-th, the auto-batching is not triggered _implicitly_ by the throughput hint.

>  - - Notice the _explicit_ notion for example "BATCH:GPU" uses the relaxed dimensions tracking, often making the auto-batching possible. For example this tricks unlocks most **detection networks**.

Reviewer 2: ```suggestion
 -  The _explicit_ notion, for example, "BATCH:GPU", uses the relaxed dimensions tracking, often making the auto-batching possible. For example, this trick unlocks most **detection networks**.
```

>  - - When "forcing" the auto-batching via the explicit device notion, make sure to validate the results for correctness.   

Reviewer 2: ```suggestion
 - - When *forcing* the auto-batching via the explicit device notion, make sure to validate the results for correctness.   
```

 - Performance improvements happen at the cost of the memory footprint growth, yet the auto-batching queries the available memory (especially for the dGPUs) and limits the selected batch size accordingly.

 
### Configuring the Automatic Batching
Following the OpenVINO convention for devices names, the *batching* device is named *BATCH*. The configuration options are as follows:

| Parameter name     | Parameter description      | Default            |             Examples                                                      |
| :---               | :---                  | :---               |:-----------------------------------------------------------------------------|

> | "AUTO_BATCH_DEVICE" | Device name to apply the automatic batching and optional batch size in brackets | N/A | "BATCH:GPU" which triggers the automatic batch size selection. Another example is device name (to apply the batching) with directly specified batch size "BATCH:GPU(4)"     |

Reviewer 2: ```suggestion
| "AUTO_BATCH_DEVICE" | Device name to apply the automatic batching and optional batch size in brackets | N/A | "BATCH:GPU" which triggers the automatic batch size selection. Another example is the device name (to apply the batching) with directly specified batch size "BATCH:GPU(4)"     |
```

| "AUTO_BATCH_TIMEOUT" | timeout value, in ms | 1000 |  you can reduce the timeout value (to avoid performance penalty when the data arrives too non-evenly) e.g. pass the "100", or in contrast make it large enough e.g. to accommodate inputs preparation (e.g. when it is serial process)     |

### Testing Automatic Batching Performance with the Benchmark_App
The `benchmark_app`, that exists in both  [C++](../../samples/cpp/benchmark_app/README.md) and [Python](../../tools/benchmark_tool/README.md) versions, is the best way to evaluate the performance of the Automatic Batching:
 -  The most straighforward way is performance hints:
- - benchmark_app **-hint tput** -d GPU -m 'path to your favorite model'
 -  Overriding the strict rules of implicit reshaping by the batch dimension via the explicit device notion:
- - benchmark_app **-hint none -d BATCH:GPU** -m 'path to your favorite model'
 -  Finally, overriding the automatically-deduced batch size as well:
- - $benchmark_app -hint none -d **BATCH:GPU(16)** -m 'path to your favorite model'

> The last example is also applicabe to the CPU and any other device that generally supports the batched execution.  

Reviewer 2: ```suggestion
The last example is also applicable to the CPU and any other device that generally supports the batched execution.  
```
